application/common/models/Method.php

```diff
@@ -179,20 +188,64 @@ public static function createAndSendVerification($userId, $type, $value)
     }
 
     /**
+     * Checks for an existing unverified method and resends verification code if found
+     * @param integer $userId
+     * @param string $type
+     * @param string $value
+     * @return Method|null
+     */
+    public static function checkForExistingAndResend($userId, $type, $value)
+    {
+        $existing = Method::find()->where([
+            'user_id' => $userId,
+            'type' => $type,
+            'verified' => 0,
+        ])->andWhere([
+            '>=', 'verification_expires', Utils::getDatetime()
+        ])->all();
+
+        /** @var Method $existMethod */
+        foreach ($existing as $existMethod) {
+            if ($existMethod->type === self::TYPE_PHONE) {
+                if (Utils::stripNonNumbers($existMethod->value) === Utils::stripNonNumbers($value)) {
+                    $existMethod->sendVerification();
+                    return $existMethod;
+                }
+            } else {
```

Comment:
To help clarify that the only "else" case (currently) is email, it may be worth doing an `elseif ($existMethod->type === self::TYPE_EMAIL) {` here, possibly with a separate "else" case to throw an error in case we ever add another type and forget to update this.